pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -110,14 +111,29 @@ func (s *Server) initCertController(args *PilotArgs) error {
 //
 // TODO: If the discovery address in mesh.yaml is set to port 15012 (XDS-with-DNS-certs) and the name
 // matches the k8s namespace, failure to start DNS server is a fatal error.
-func (s *Server) initDNSCerts(hostname, namespace string) error {
+func (s *Server) initDNSCerts(hostname, customHostname, namespace string) error {
 	parts := strings.Split(hostname, ".")
 	if len(parts) < 2 {
 		return fmt.Errorf("invalid hostname %s, should contain at least service name and namespace", hostname)
 	}
+
 	// Names in the Istiod cert - support the old service names as well.
 	// The first is the recommended one, also used by Apiserver for webhooks.
 	names := []string{hostname}
+
+	// validate and append custom hostname if there is any
+	if customHostname != "" {
+		partsCustom := strings.Split(customHostname, ".")
```

Comment:
This is probably already done by the signing package. 

Please use the existing code in security to pass the SANs - and the API, it has been 
reviewed and approved by security WG AFAIK (chiron proposal).

Reply:
happy to reuse and take wildcard too.  @costinm or @myidpt could you give me a pointer of the code?  I can't find it.

Reply:
We do have implementation in 1.5 but it's removed on master: https://github.com/istio/istio/blob/release-1.5/security/pkg/k8s/controller/customdnsname.go

Reply:
Thanks, I checked, there is not much validation for hostname.

Reply:
I don't think we need too much validation - in practice the client or browser will reject the cert.

Just want to make sure we don't accidentally break wildcard certs ( *.foo.com ). The code signing the certificate should do any required validation.